src/main/java/appeng/tile/powersink/AEBasePoweredTile.java

```diff
@@ -123,34 +123,20 @@ public final double injectAEPower( double amt, final Actionable mode )
 			return 0;
 		}
 
-		if( mode == Actionable.SIMULATE )
-		{
-			final double fakeBattery = this.getInternalCurrentPower() + amt;
+		final double required = this.getAEMaxPower() - this.getAECurrentPower();
+		final double insertable = Math.min( required, amt );
 
-			if( fakeBattery > this.getInternalMaxPower() )
-			{
-				return fakeBattery - this.getInternalMaxPower();
-			}
-
-			return 0;
-		}
-		else
+		if( mode == Actionable.MODULATE )
 		{
-			if( this.getInternalCurrentPower() < 0.01 && amt > 0.01 )
+			if( this.getInternalCurrentPower() < 0.01 && insertable > 0.01 )
 			{
 				this.PowerEvent( PowerEventType.PROVIDE_POWER );
 			}
 
-			this.setInternalCurrentPower( this.getInternalCurrentPower() + amt );
-			if( this.getInternalCurrentPower() > this.getInternalMaxPower() )
-			{
-				amt = this.getInternalCurrentPower() - this.getInternalMaxPower();
-				this.setInternalCurrentPower( this.getInternalMaxPower() );
-				return amt;
-			}
-
-			return 0;
+			this.setInternalCurrentPower( this.getInternalCurrentPower() + insertable );
 		}
+
+		return amt - insertable;
 	}
 
 	protected void PowerEvent( final PowerEventType x )
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00100)